mlrun/model.py

```diff
@@ -56,22 +60,115 @@ def _verify_dict(param, name, new_type=None):
             return new_type.from_dict(param)
         return param
 
-    def to_dict(self, fields=None, exclude=None):
-        """convert the object to a python dictionary"""
+    def to_dict(
+        self, fields: list = None, exclude: list = None, strip: bool = False
+    ) -> dict:
+        """
+        convert the object to a dict
+        :param fields:  A list of fields to include in the dictionary. If not provided, the default value is taken
+        from `self._dict_fields` or from the object __init__ params.
+        First we run over the fields and serialize them, then we run over the fields and enrich them.
+        :param exclude: A list of fields to exclude from the dictionary. If not provided, the default value is an
+         empty list and the object's `_fields_to_exclude_for_serialization` attribute is appended to it.
+        :param strip:  If True, the object's `_fields_to_strip` attribute is appended to the exclude list.
+        :return: A dictionary representation of the object.
```

Comment:
I'm missing an explanation for the role of the `_fields_to_exclude_for_enrichment` property. Please add an explanation of what it does.